File: src/main/java/com/VikingLabs/MielIA/DTOs/MedicalStudyDTO.java
package com.VikingLabs.MielIA.DTOs;

import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.constraints.*;
import lombok.Data;

import java.util.UUID;

@Data
@Tag(name="DTO", description = "Data Transfer Objects")
@Schema(name = "MedicalStudyDTO", description = "DTO para creación y visualización de estudios médicos")
public class MedicalStudyDTO {

    @Schema(description = "ID del paciente", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotNull(message = "El ID del paciente es obligatorio")
    private UUID patientId;

    @Schema(description = "ID del médico", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotNull(message = "El ID del médico es obligatorio")
    private UUID doctorId;

    @Schema(description = "ID del técnico (opcional)")
    private UUID technicianId;

    @Schema(description = "Código de acceso de 4 caracteres", requiredMode = Schema.RequiredMode.REQUIRED)
    @Size(min = 4, max = 4, message = "El código debe tener exactamente 4 caracteres")
    private String accessCode;

    @Schema(description = "Datos clínicos en formato JSON", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "Los datos clínicos son obligatorios")
    private String clinicalData;
}

File: app/infrastructure/db/DTOs/medical_study_dto.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from uuid import UUID
from .user_dto import DoctorInfoDTO, PatientInfoDTO
from .base_dto import BaseDTO

class MedicalStudyCreateDTO(BaseDTO):
    access_code: str = Field(..., description="Código de acceso único")
    doctor_id: UUID = Field(..., description="ID del médico solicitante")
    patient_id: UUID = Field(..., description="ID del paciente")
    technician_id: Optional[UUID] = Field(None, description="ID del técnico")
    clinical_data: Optional[str] = Field(None, description="Datos clínicos iniciales")

class MedicalStudyUpdateDTO(BaseDTO):
    access_code: Optional[str] = None
    doctor_id: Optional[UUID] = None
    patient_id: Optional[UUID] = None
    technician_id: Optional[UUID] = None
    clinical_data: Optional[str] = None
    status: Optional[str] = None
    ml_results: Optional[str] = None
    csv_file_id: Optional[UUID] = None

class MedicalStudyResponseDTO(BaseDTO):
    id: UUID
    access_code: str
    status: str
    creation_date: datetime
    ml_results: Optional[str]
    patient: PatientInfoDTO
    doctor: Optional[DoctorInfoDTO] = None